src/main/kotlin/fi/metatavu/megasense/dataportal/users/UsersController.kt

```diff
@@ -254,23 +240,79 @@ class UsersController {
         return result
     }
 
+    /**
+     * Builds ordered list of pollutant names to be later used as headers for csv file
+     *
+     * @param exposureInstances all exposure instances being exported
+     * @return list of pollutant names
+     */
+    private fun orderHeaderNames(exposureInstances: List<ExposureInstance>): List<String> {
+        val exposureInstancePollutants = mutableListOf<Pollutant>()
+        exposureInstances.map { instance ->
+            instance.pollutionEntries?.forEach { pollutionEntry ->
+                if (pollutionEntry.pollutant != null && !exposureInstancePollutants.contains(pollutionEntry.pollutant)) {
+                    exposureInstancePollutants.add(pollutionEntry.pollutant!!)
+                }
+            }
+        }
+        exposureInstancePollutants.sortBy { p -> p.name }
+        return exposureInstancePollutants.map { pollutant -> pollutant.displayName }.filterNotNull()
+    }
+
+    /**
+     * Arranges the pollution values in the same order with csv headers, filling the empty spaces with zeroes,
+     * building one record for csv file
+     *
+     * @param pollutionValues list of pollution entries for the record
+     * @param pollutionHeaders list of pollutants in csv header
+     * @return pollution entries ordered correctly
+     */
+    private fun orderRecordValues(
+        pollutionValues: List<PollutionEntry>?,
+        pollutionHeaders: List<String>
+    ): TreeMap<Int, Float?> {
+        //put pollution values in same order as pollutants in headers
+        val pollutionEntries = hashMapOf<Int, Float?>()
+        pollutionValues?.forEach { pollutionEntry ->
+            val pollutantName = pollutionEntry.pollutant?.displayName
+            val indexOfPollutantInHeader = pollutionHeaders.indexOf(pollutantName)
+            pollutionEntries[indexOfPollutantInHeader] = pollutionEntry.value
+        }
+
+        //fill missing pollution values with 0
+        val sortedPollutionValues = TreeMap<Int, Float?>(pollutionEntries)
+        if (sortedPollutionValues.size < pollutionHeaders.size) {
+            for (i in pollutionHeaders.indices) {
+                if (!sortedPollutionValues.containsKey(i)) {
+                    sortedPollutionValues[i] = 0f
+                }
+            }
+        }
+        return sortedPollutionValues
+    }
+
     /**
      * Writes routes to a csv string
      *
      * @param routes routes to write
      *
      * @return csv string
```

Comment:
```suggestion
     * @return csv string
```
No empty line between @param and @return needed